app/src/main/java/mn/devfest/sessions/SessionsFragment.java

```diff
@@ -44,35 +51,67 @@
  * @author bherbst
  * @author pfuentes
  */
-public class SessionsFragment extends Fragment implements DevFestDataSource.DataSourceListener, OnCategoryFilterSelectedListener {
+public class SessionsFragment extends Fragment implements DevFestDataSource.DataSourceListener, OnCategoryFilterSelectedListener, SessionViewHolder.ToggleInScheduleListener {
     private static final String PREF_KEY_AUTOHIDE = "autohide_past_sessions";
 
     // TODO this shouldn't be static so we can localize
-    private static final String ALL_CATEGORY = "All";
-    private static final int MINUTES_BEFORE_ENDTIME_TO_SHOW_SESSION_FEEDBACK = 20;
+    public static final String ALL_CATEGORY = "All";
+    public static final int MINUTES_BEFORE_ENDTIME_TO_SHOW_SESSION_FEEDBACK = 20;
+    public static final String DEVFEST_2017_KEY = "devfest2017";
+    public static final String SESSIONS_CHILD_KEY = "schedule";
+    public static final String SPEAKERS_CHILD_KEY = "speakers";
 
     @Bind(R.id.session_list_recyclerview)
     RecyclerView mSessionRecyclerView;
 
     @Bind(R.id.loading_progress)
     ProgressBar mLoadingView;
 
-    private SessionListAdapter mAdapter;
-
-    private DevFestDataSource mDataSource;
     private SharedPreferences mPreferences;
     private Subscription mDataUpdateSubscription;
 
     private boolean mAutohidePastSessions;
     private List<Session> mAllSessions;
     private Set<String> mAllCategories;
     private String mCategoryFilter;
+    private DatabaseReference mFirebaseDatabaseReference;
+    private SessionListAdapter mAdapter;
+    private LinearLayoutManager mLayoutManager;
+
+    @Override
+    public boolean onToggleScheduleButtonClicked(Session session) {
+        //TODO handle toggling schedule
+        return false;
+    }
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setHasOptionsMenu(true);
 
+        mFirebaseDatabaseReference = FirebaseDatabase.getInstance().getReference();
```

Comment:
Inject the Firebase database using Dagger when you need it. Assuming that the app isn't set up with Dagger for nothing.